src/gpuarray_reduction.c

```diff
@@ -256,113 +801,492 @@ static int   maxandargmaxCheckargs              (maxandargmax_ctx*  ctx){
 	ctx->nds = ctx->src->nd;
 	ctx->ndr = ctx->reduxLen;
 	ctx->ndd = ctx->nds - ctx->ndr;
+	strb_ensure(&ctx->s, 5*1024);
+
 
-	return ctx->ret;
+
+	return reduxSelectModel(ctx);
```

Comment:
I'm not sure.  Something like `reduxCheckArgsAndRest()`?  I don't like it very much ...